packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx

```diff
@@ -226,13 +226,24 @@ export const TreeItem: ComponentWithAs<'div', TreeItemProps> & FluentComponentSt
   };
 
   const handleTitleClick = e => {
-    if (hasSubtree && e.target === e.currentTarget) {
+    if (selectable) {
+      if (hasSubtree) {
+        // need to know if click happened on selection indicator or title itself
+        if (e?.target?.className?.includes(treeTitleSlotClassNames.indicator)) {
```

Comment:
i.e. add an handler to `TreeTitle` for the indicator to call `toggleItemSelect` but also call `stopPropagation`. This way you have correct event encapsulation.

Reply:
Heyyy Ling so I was trying to add onClick to the selection indicator, and it turns out a bit more troubling then I think, see comments in the snippet below:
```ts
// I'm adding this in TreeTitle.tsx
  const onClickSelectIndicator = e => {
    if (selectable) {
      e.preventDefault();
      toggleItemSelect(e, id);  // The id here, is not a prop of TreeTitle :( we would have to figure out a way to pass it from treeItem
    }
    _.invoke(props, 'onTitleClick', e, props);
  };
```

Reply:
Could we just pass `id` to selection indicator?

Reply:
@assuncaocharles I'm not sure how? Because selection indicator is in `treeTitle`, and there's no id in `treeTitle`

Reply:
Well, drilling down the prop?

Reply:
Does it not make sense to add `id` as a prop to `TreeTitle` ? If not you could consider creating this callback in `TreeItem` and make `TreeTitle` accept the callback as prop

Reply:
@ling1726 I'm not sure, because when you create a tree component using shorthand, it goes:
```ts
const items = [
  {
    id: 'tree-item-111',
    title: 'Jaime',
  },
];
const TreeExample = () => <Tree items={items} />;
```
So technically id is not part of the title. 
But I'm open to discussion, maybe we can brought it up tomorrow on stand up?
(Once again separation of treeItem/treeTitle screws us up :D )

Reply:
isn't the `selectionIndicator` just a shorthand box prop ? can you pass an onClick handler through there ?

Reply:
That wouldn't work @ling1726 if we pass something for `selectionIndicator` we gonna have to pass `content` as well if I am not mistaken